File: pizza_flutter/lib/models/product.dart
class Product {
  final String id;
  final String name;
  final String description;
  final double price;
  final String imageUrl;
  final String categoryId;
  final String category; // = categoryName từ BE (giữ tên "category" cho các màn hình cũ)

  Product({
    required this.id,
    required this.name,
    required this.description,
    required this.price,
    required this.imageUrl,
    required this.categoryId,
    required this.category,
  });

  factory Product.fromJson(Map<String, dynamic> json) => Product(
    id: json['id']?.toString() ?? '',
    name: json['name'] ?? '',
    description: json['description'] ?? '',
    price: (json['price'] as num).toDouble(),
    imageUrl: json['imageUrl'] ?? '',
    categoryId: json['categoryId']?.toString() ?? '',
    category: json['categoryName'] ?? '',
  );
}

File: backend/src/Services/Product/PizzaApp.Product.Core/Entities/Product.cs
namespace PizzaApp.Product.Core.Entities;

public class Product
{
    public string Id { get; set; } = string.Empty;
    public string Name { get; set; } = string.Empty;
    public string Description { get; set; } = string.Empty;
    public decimal Price { get; set; }
    public string ImageUrl { get; set; } = string.Empty;
    public string CategoryId { get; set; } = string.Empty;
    public string CategoryName { get; set; } = string.Empty; // denormalized từ Category service
    public bool IsAvailable { get; set; } = true;
}
